File: Leetcode/Leetcode-DSA-course/Arrays and Strings/Sliding window Technique/2-Max sum subarray of size k/main.js
function getMaxSum(nums, k) {
    var curr = 0;
    for (var i = 0; i < k; i++)
        curr += nums[i];
    var ans = curr;
    for (var i = k; i < nums.length; i++) {
        curr += nums[i] - nums[i - k];
        ans = Math.max(ans, curr);
    }
    return ans;
}
var nums = [3, -1, 4, 12, -8, 5, 6];
var k = 4;
console.log(getMaxSum(nums, k));

File: Leetcode/Leetcode-DSA-course/Arrays and Strings/Sliding window Technique/2-Max sum subarray of size k/main.cpp
// Given an integer array nums and an integer k, find the sum of the subarray with the largest sum whose length is k.

#include <bits/stdc++.h>
using namespace std;

int getMaxSum(vector<int>& nums, int k) {
    int curr = 0;
    for (int i = 0;i < k;i++)
        curr += nums[i];
    int ans = curr;
    for (int i = k;i < nums.size();i++) {
        curr += nums[i] - nums[i - k];
        ans = max(ans, curr);
    }

    return ans;
}

int main() {
    vector<int> nums = { 3,-1,4,12 ,-8,5,6 };
    int k = 4;
    cout << getMaxSum(nums, k) << endl;
}
